pkg/controller/utils/utils.go

```diff
@@ -92,3 +93,20 @@ func DeleteFinalizer(object metav1.Object, finalizer string) {
 	finalizers.Delete(finalizer)
 	object.SetFinalizers(finalizers.List())
 }
+
+// GetKubeClient creates a new Kubernetes dynamic client.
+func GetKubeClient(scheme *runtime.Scheme) (client.Client, error) {
```

Comment:
@csrwng if it's ok i'd propose keeping this function, it could be generally useful.